File: cpp/num_20.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> a;
        for (const auto &i : s)
        {
            switch (i)
            {
            case '(':
            case '[':
            case '{':
                a.push(i);
                break;
            case ')':
                if (!a.empty() && a.top() == '(')
                {
                    a.pop();
                    break;
                }
                else
                {
                    return false;
                }
            case ']':
                if (!a.empty() && a.top() == '[')
                {
                    a.pop();
                    break;
                }
                else
                {
                    return false;
                }
            case '}':
                if (!a.empty() && a.top() == '{')
                {
                    a.pop();
                    break;
                }
                else
                {
                    return false;
                }
            default:
                break;
            }
        }
        return a.empty() ? true : false;
    }
};

File: py/num_20.py
class Solution:
    def isValid(self, s: str) -> bool:
        a = []
        for i in s:
            if i == '(' or i == '[' or i == '{':
                a.append(i)
            elif i == ')':
                if a and a[-1] == '(':
                    a.pop()
                else:
                    return False
            elif i == ']':
                if a and a[-1] == '[':
                    a.pop()
                else:
                    return False
            elif i == '}':
                if a and a[-1] == '{':
                    a.pop()
                else:
                    return False
            else:
                pass
        if not a:
            return True
        else:
            return False
